### ISSUE TITLE
[Product BUG]: Admin | Sync Template v3 (Apps) Failed

### ISSUE THREAD

**Issue by apoorvakasbekar** [microsoft/powerapps-tools#830]
Admin | Sync Template v3 (Apps) failed with the below error:
**The API 'powerappsforadmins' returned an invalid response for workflow operation 'Get_Apps_as_Admin' of type 'OpenApiConnection'. Error details: 'The API operation 'Get-AdminApps' requires the property 'body/value/30/properties/connectionReferences/0/iconUri' to be of type 'String' but is of type 'Null'.'**

Solution - Core
Version - 1.98
Flow - Admin | Sync Template v3 (Apps)

![image](https://user-images.githubusercontent.com/69908205/111015558-61f75480-83cf-11eb-954a-d3411b3fb56a.png)

**Comment by JeneferM-MSFT** [comment]
Can you please check the following?

1) rerun the flow and see if it repros a second time, or if this was internittent. 
1) create a separate flow which tests just this call for your default environment. Here is what that would look like
![image](https://user-images.githubusercontent.com/62659043/111073047-66d50a80-849a-11eb-93de-bb2b4f56fcd3.png)

**Comment by apoorvakasbekar** [comment]
![image](https://user-images.githubusercontent.com/69908205/111096612-f4c9f780-8565-11eb-8bfa-9cdaede6888d.png)

**Failed with the below error:** 
ResponseSwaggerSchemaValidationFailure. The API 'powerappsforadmins' returned an invalid response for workflow operation 'Get_Apps_as_Admin' of type 'OpenApiConnection'. Error details: 'The API operation 'Get-AdminApps' requires the property 'body/value/30/properties/connectionReferences/0/iconUri' to be of type 'String' but is of type 'Null'.'

**Comment by apoorvakasbekar** [comment]
NOTE: If I select the CoE_Environment (org66c50f51) it works. It fails when I select the Default Environment 

![image](https://user-images.githubusercontent.com/69908205/111173867-f975c680-85cc-11eb-9753-12222019f571.png)

**Comment by paulvandyck** [comment]
I am getting this exact same error in our environment.

**Comment by JeneferM-MSFT** [comment]
Paul, do you also repro outside of the context of the toolkit as shown above?

**Comment by apoorvakasbekar** [comment]
@JeneferM-MSFT thank you for your assistance with all the CoE related issues. Any luck with this one ? 

This one is very important for us.

**Comment by JeneferM-MSFT** [comment]
Not yet, I believe you have hit a product bug and I was going to try and investigate a little further before just directing you to support, but perhaps you should just create a ticket while I also try and investigate. 

For best traction with support, just refer to the flow you created outside of the context of the toolkit so that they do not think the error is specific to CoE and cause delay. 

Here is what we know:
1) This repro's outside of the toolkit, so the connector is just not working for your default environment
2) This isn't just happening to you,  there are two customers now here that are hitting this.
3) This does not repro for my default environment in any of my test tenants.

**Comment by apoorvakasbekar** [comment]
The support team identified that a few corrupt apps in my default environment were causing an issue with 'Get Apps as Admin' in for the Default Environment. I deleted those apps and ran the flow and it was able to run 'Get Apps as Admin'

However, now I have a new error: 

![image](https://user-images.githubusercontent.com/69908205/111783043-ca6a9800-88df-11eb-956f-fe16bd93f1ca.png)

Schema: 

{
  "type": "array",
  "items": {
    "type": "object",
    "properties": {
      "iconUri": {
        "type": "string"
      },
      "displayName": {
        "type": "string"
      },
      "apiTier": {
        "type": "string"
      },
      "isOnPremiseConnection": {
        "type": "boolean"
      },
      "isCustomApiConnection": {
        "type": "boolean"
      },
      "bypassConsent": {
        "type": "boolean"
      },
      "id": {
        "type": "string"
      }
    },
    "required": [
      "iconUri",
      "displayName",
      "apiTier",
      "isOnPremiseConnection",
      "isCustomApiConnection",
      "bypassConsent",
      "id"
    ]
  }
}

**Comment by JeneferM-MSFT** [comment]
Glad to hear the first issue is resolved. It seems like you might now be seeing #812 . Can you please take a look and add your notes there? There is a suggested fix that I need someone who repros to try. 

Closing this out since the original issue is fixed. 

Thanks!

**Comment by paulvandyck** [comment]
I didn't see the question for me until today.    I think i was being asked if I could reproduce this error outside the context of the tool kit.   The answer is I don't know.    We only see this error from the Admin | Sync Template v3 flow that is part of the kit.    We had a support ticket with Microsoft open for the past week, and 2 days ago they closed the ticket, noting that they don't support the PP COE starter kit, and pointed us to this github.   I looked at the github and noted this issue has the same exact error message and commented.      The original poster has noted the support team found corrupt apps and they deleted them, and the original posters issue was resolved.    We've not gotten support help to identify corrupt apps.

In the end, we've deleted the Power Platform COE starter kit environment and decided that we can't use it as a tool to help us govern or monitor our Power Platform.  After 4-5 weeks of trying to set it up and being handed off from different Microsoft teams , we no longer have the time to make this work.
